sphinx/environment.py

```diff
@@ -961,6 +962,29 @@ def process_downloads(self, docname, doctree):
 
     def process_images(self, docname, doctree):
         """Process and rewrite image URIs."""
+        def collect_candidates(imgpath, candidates):
+            globbed = {}
+            for filename in glob(imgpath):
+                new_imgpath = relative_path(path.join(self.srcdir, 'dummy'),
+                                            filename)
+                if filename.lower().endswith('.pdf'):
```

Comment:
Also, while we're here, the whole *file extension to mime type* mapping could be refactored into a mapping, instead of a collection of ifs.